met-api/migrations/versions/e39d3e01dc22_create_widget_tables.py

```diff
@@ -0,0 +1,80 @@
+"""Create widget tables
+
+Revision ID: e39d3e01dc22
+Revises: ac4a505ed1e3
+Create Date: 2022-10-18 08:30:48.529328
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = 'e39d3e01dc22'
+down_revision = 'ac4a505ed1e3'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('contact',
+    sa.Column('id', sa.Integer(), autoincrement=True, nullable=False),
+    sa.Column('name', sa.String(length=50), nullable=True),
+    sa.Column('role', sa.String(length=50), nullable=True),
+    sa.Column('email', sa.String(length=50), nullable=True),
+    sa.Column('phone_number', sa.String(length=50), nullable=True),
+    sa.Column('address', sa.String(length=50), nullable=True),
+    sa.Column('bio', sa.String(length=500), nullable=True),
+    sa.Column('created_date', sa.DateTime(), nullable=False),
+    sa.Column('updated_date', sa.DateTime(), nullable=False),
+    sa.Column('created_by', sa.String(length=50), nullable=False),
+    sa.Column('updated_by', sa.String(length=50), nullable=False),
+    sa.PrimaryKeyConstraint('id')
+    )
+    widget_type_table = op.create_table('widget_type',
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('name', sa.String(length=50), nullable=False),
+    sa.Column('description', sa.String(length=200), nullable=True),
+    sa.PrimaryKeyConstraint('id'),
+    sa.UniqueConstraint('name')
+    )
+    op.create_table('widget',
+    sa.Column('id', sa.Integer(), autoincrement=True, nullable=False),
+    sa.Column('widget_type_id', sa.Integer(), nullable=True),
+    sa.Column('engagement_id', sa.Integer(), nullable=True),
+    sa.Column('created_date', sa.DateTime(), nullable=False),
+    sa.Column('updated_date', sa.DateTime(), nullable=False),
+    sa.Column('created_by', sa.String(length=50), nullable=False),
+    sa.Column('updated_by', sa.String(length=50), nullable=False),
+    sa.ForeignKeyConstraint(['engagement_id'], ['engagement.id'], ondelete='CASCADE'),
+    sa.ForeignKeyConstraint(['widget_type_id'], ['widget_type.id'], ondelete='CASCADE'),
+    sa.PrimaryKeyConstraint('id'),
+    sa.UniqueConstraint('widget_type_id', 'engagement_id', name='unique_widget_type')
+    )
+    op.create_table('widget_item',
+    sa.Column('id', sa.Integer(), autoincrement=True, nullable=False),
+    sa.Column('widget_data_id', sa.Integer(), nullable=False),
```

Comment:
We have to add comment for widget_data_id which will go to DB..
I wanna suggest a better name for it so that the link between contact(or some other table) is conveyed in the bame itself,but i think widget_data_id is much better than what i am coming up with...

Reply:
Added comment. Widget_data_id is the best name I came up with